File: Models/Word.cs
struct Word
{
    public Word(string simplified, string traditional, string sounds, string pinyin, string definition)
    {
        Simplified = simplified;
        Traditional = traditional;
        Sounds = sounds;
        Pinyin = pinyin;
        Definition = definition;
    }

    public string Simplified { get; set; }
    public string Traditional { get; set; }
    public string Sounds { get; set; }
    public string Pinyin { get; set; }
    public string Definition { get; set; }
}

File: src/app/word.ts
const minute = 60 * 1000;
const day = 24 * 60 * minute;

export class Word {
  interval: number = minute;
  due?: number;

  constructor(
    public simplified: string,
    public traditional: string,
    public sounds: string,
    public pinyin: string,
    public definition: string
  ) {}

  gotIt(): void {
    this.due = Date.now() + this.interval;
    if (this.interval == 4 * minute) this.interval = day;
    if (this.interval <= 32 * day) this.interval *= 2;
  }

  again(): void {
    this.interval = minute;
    this.due = Date.now() + this.interval;
  }
}
